src/items.cpp

```diff
@@ -326,9 +326,33 @@ bool Items::loadFromXml()
 			parseItemNode(itemNode, id++);
 		}
 	}
+
+	buildInventoryList();
 	return true;
 }
 
+void Items::buildInventoryList()
+{
+	inventory.reserve(items.size());
+	for(const auto& type: items) {
```

Comment:
You forgot to add space between ``for`` and ``(``. You should also rename ``type`` to ``itemType`` (weird for a ``vector`` used to hold ``ItemType``s to be named loosely as ``items``)